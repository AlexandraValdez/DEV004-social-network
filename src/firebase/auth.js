import {
  getAuth,
  createUserWithEmailAndPassword,
  signInWithEmailAndPassword,
  signOut,
  GoogleAuthProvider,
  FacebookAuthProvider,
  signInWithPopup,
  sendPasswordResetEmail,
} from 'firebase/auth';
import { app } from './config.js';

export const auth = getAuth(app);

// eslint-disable-next-line max-len
export const addUser = (email, password) => createUserWithEmailAndPassword(auth, email, password);

export const userInfo = (email, password) => signInWithEmailAndPassword(auth, email, password);

export const google = new GoogleAuthProvider();

export const facebook = new FacebookAuthProvider();

export const loginGoogle = () => signInWithPopup(auth, google);
export const loginFacebook = () => signInWithPopup(auth, facebook);
export const recoverPassword = (email) => sendPasswordResetEmail(email);
export const logOut = () => signOut(auth);
export const currentUser = () => auth().currentUser.uid;
