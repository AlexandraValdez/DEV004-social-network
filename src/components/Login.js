import { async } from '@firebase/util';
import { signInWithEmailAndPassword } from 'firebase/auth';
import { getDatabase, update, ref } from 'firebase/database';
import { auth, database } from '../app/firebase.js';
import { showMessage } from '../app/showMessage.js';
import { onNavigate } from '../main';
/* Para que este disponoble en otro lado export */
export const Login = () => {
  /* UN CONTENEDOR Q CONTENGA A LOS BOTONES */
  const div = document.createElement('div');
  const title = document.createElement('h2');
  /* crear botones createElement */
  const button = document.createElement('button');
  const buttonBack = document.createElement('button');
  const buttonProfile = document.createElement('button');
  const section = document.createElement('div');
  section.innerHTML = `
    <div class="wrapper">
      <div class="img-container">
        <img src="./images/Sonder-icon.png" alt="">
      </div>
      <h1>Log In</h1>
      <p class="text" >Welcome back you've been missed!</p>
      
      <form id="login-form" class="login-form">
        <label for="email"></label>
        <input type="email" id="login-email" class="" placeholder="Email" required>
        <input type="password" id="login-password"  class="" placeholder="Password" required>
        <p class="recover"> <a href="" >Forgot password?</a></p> 
      
      <button type="submit" id="log" class="submit">Log in</button>
      </form> 
      <p class="or">or</p>
        <button type="submit" class="facebook">Continue with Facebook</button>
        <button type="submit" class="google">Continue with Google</button>
       
        <p class="sign-up">
          Dont have an account yet? <a href="/signup">Sign Up</a>
        </p> 
   </div>
  `;

  /* FIREBASE FORM */
  const loginForm = section.querySelector('#login-form');

  loginForm.addEventListener('submit', async (e) => {
    e.preventDefault();
    const email = loginForm['login-email'].value;
    const password = loginForm['login-password'].value;
    // eslint-disable-next-line no-console
    console.log(email, password);

    try {
      // eslint-disable-next-line max-len
      const userCredentials = await signInWithEmailAndPassword(auth, email, password); // le paso los parametros que quiero auth
      const dt = new Date();
      update(ref(database, `users/${userCredentials.user.uid}`), {
        last_login: dt,
      });
      onNavigate('/home');

      showMessage(`Welcome ${userCredentials.user.email}`, 'success');
    } catch (error) {
      if (error.code === 'auth/wrong-password') {
        showMessage('Wrong password', 'error'); // despues de la coma viene el tipo (estilo que le cambia el color al msg)
      } else if (error.code === 'auth/user-not-found') {
        showMessage('User not found', 'error');
      } else if (error.code) {
        showMessage(error.message, 'error');
      }
    }
  });

  /* AGREGAR TEXTO A LOS BOTONES textContent */
  button.textContent = 'Entrar';
  buttonBack.textContent = 'Regresa';
  buttonProfile.textContent = 'Profile';
  title.textContent = 'Inicia secion LOGIN';

  button.addEventListener('click', () => {
    onNavigate('/home');
  });
  buttonBack.addEventListener('click', () => {
    onNavigate('/');
  });
  buttonProfile.addEventListener('click', () => {
    onNavigate('/profile');
  });

  /* INSERTA append */
  div.append(title, buttonProfile, button, buttonBack, section);

  return div;
};
