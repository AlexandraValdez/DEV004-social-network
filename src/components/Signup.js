import { createUserWithEmailAndPassword, GoogleAuthProvider, signInWithPopup, FacebookAuthProvider } from 'firebase/auth';

import { getDatabase, set, ref } from 'firebase/database';
import { onNavigate } from '../main';
import { auth, database } from '../app/firebase.js';
import { showMessage } from '../app/showMessage.js';
/* Para que este disponoble en otro lado export */
export const signUp = () => {
  /* UN CONTENEDOR Q CONTENGA A LOS BOTONES */
  const div = document.createElement('div');
  const title = document.createElement('h2');
  /* crear botones createElement */
  // const button = document.createElement("button");
  const buttonBack = document.createElement('button');
  // const inputEmail = document.createElement("input");
  // const inputPass = document.createElement("input");
  const section = document.createElement('div');
  section.innerHTML = `
    <div class="wrapper">
      <div class="img-container">
        <img src="./images/Sonder-icon.png" alt="">
      </div>
      <h1>Sign Up</h1>
      <p class="text-sign">Find and share inspiration all around the world!</p>
      <form id="signup-form" >
        <input id="singup-name" type="text" class="" placeholder="Name">
        <input id="singup-lastname" type="text" class="" placeholder="Last Name">
        <input id="singup-email" type="email" class="" placeholder="Email">
        <input id="singup-password" type="password" class="" placeholder="Password">
        <input type="password" class="" placeholder="Confirm Password">
      <button type="submit" class="submit">Sign Up</button>
      </form>  
      <p class="or">or</p>
        <button id="fb-login" type="button" class="facebook">Continue with Facebook</button>
        <button id="googleLogin" type="button" class="google">Continue with Google</button>
        <p class="sign-up">
          Already have an account yet? <a href="/login">Log in</a>
        </p> 
    </div>
    `;
  /* FIREBASE FORM */
  const signupForm = section.querySelector('#signup-form');
  signupForm.addEventListener('submit', async (e) => {
    e.preventDefault();
    const username = signupForm['singup-name'].value;
    const lastname = signupForm['singup-lastname'].value;
    const email = signupForm['singup-email'].value;
    const password = signupForm['singup-password'].value;
    // eslint-disable-next-line no-console
    console.log(username, lastname, email, password);
    try {
      // eslint-disable-next-line max-len
      const userCredentials = await createUserWithEmailAndPassword(
        auth,
        email,
        password,
      ); // le paso los parametros que quiero auth
      console.log(userCredentials.user);
      set(ref(database, `users/${userCredentials.user.uid}`), {
        name: username,
        lastname,
        email,
      });
      onNavigate('/profile');
      showMessage(`Welcome ${userCredentials.user.email}`, 'success');
    } catch (error) {
      if (error.code === 'auth/email-already-in-use') {
        showMessage('Email already in use', 'error'); // despues de la coma viene el tipo (estilo que le cambia el color al msg)
      } else if (error.code === 'auth/invalid-email') {
        showMessage('Invalid email', 'error');
      } else if (error.code === 'auth/weak-password') {
        showMessage('Weak password', 'error');
      } else if (error.code) {
        showMessage(error.message, 'error');
      }
    }
  });

  /* Google log-in */
  const googleBtn = section.querySelector('#googleLogin');
  googleBtn.addEventListener('click', async (e) => {
    e.preventDefault();
    const provider = new GoogleAuthProvider();

    try {
      const credentials = await signInWithPopup(auth, provider);
      console.log(credentials);
      onNavigate('/profile');
      showMessage(`Welcome ${credentials.user.displayName}`, 'success');
    } catch (error) {
      if (error.code === 'auth/email-already-in-use') {
        showMessage('Email already in use', 'error'); // despues de la coma viene el tipo (estilo que le cambia el color al msg)
      } else if (error.code === 'auth/invalid-email') {
        showMessage('Invalid email', 'error');
      } else if (error.code === 'auth/weak-password') {
        showMessage('Weak password', 'error');
      } else if (error.code) {
        showMessage(error.message, 'error');
      }
    }
  });

  /* Facebook log-in */
  const facebookbtn = section.querySelector('#fb-login');
  facebookbtn.addEventListener('click', async (e) => {
    e.preventDefault();
    const provider = new FacebookAuthProvider();

    try {
      const credentials = await signInWithPopup(auth, provider);
      console.log(credentials);
      onNavigate('/profile');
      showMessage(`Welcome ${credentials.user.displayName}`, 'success');
    } catch (error) {
      if (error.code) {
        showMessage(error.message, 'error');
      }
    }
  });
  // button.textContent = "crear cuenta";
  buttonBack.textContent = 'Regresa';
  // title.textContent = "Registro de nueva cuenta";

  /* evento a boton */
  // button.addEventListener("click", () => {
  //   onNavigate("/");
  // });
  buttonBack.addEventListener('click', () => {
    onNavigate('/');
  });

  /* INSERTA append */
  div.append(title, buttonBack, section);

  return div;
};
// inputEmail, inputPass, button,
