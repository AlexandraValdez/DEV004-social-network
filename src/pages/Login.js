import { async } from '@firebase/util';
import { update, ref } from 'firebase/database';
import { loginGoogle, loginFacebook, newUser } from '../firebase/auth.js';
// import { auth, database } from "../firebase/config.js";
import { showMessage } from '../components/showMessage.js';
import { onNavigate } from '../main';
/* Para que este disponoble en otro lado export */
export const Login = () => {
  /* UN CONTENEDOR Q CONTENGA A LOS BOTONES */
  const div = document.createElement('div');
  const title = document.createElement('h2');
  /* crear botones createElement */
  const button = document.createElement('button');
  const buttonBack = document.createElement('button');
  const buttonProfile = document.createElement('button');
  const section = document.createElement('div');
  section.innerHTML = `
    <div class="wrapper">
      <div class="img-container">
        <img src="./images/Sonder-icon.png" alt="">
      </div>
      <h1>Log In</h1>
      <p class="text" >Welcome back you've been missed!</p>
      
      <form id="login-form" class="login-form">
        <label for="email"></label>
        <input type="email" id="login-email" class="" placeholder="Email" required>
        <input type="password" id="login-password"  class="" placeholder="Password" required>
        <p class="recover"> <a href="" >Forgot password?</a></p> 
        <button type="submit" id="log" class="submit">Log in</button>
      </form> 
      <p class="or">or</p>
        <button id="fb-login" type="button" class="submit facebook">Continue with Facebook</button>
        <button id="googleLogin" type="button" class="submit google">Continue with Google</button>
        <p class="p-log">
          Dont have an account yet? <a href="/signup">Sign Up</a>
        </p> 
   </div>
  `;

  /* FIREBASE FORM */
  const loginForm = section.querySelector('#login-form');

  loginForm.addEventListener('submit', async (e) => {
    e.preventDefault();
    const email = loginForm['login-email'].value;
    const password = loginForm['login-password'].value;
    // eslint-disable-next-line no-console
    console.log(email, password);

    try {
      newUser(email, password);
      /* const dt = new Date();
      update(ref(database, `users/${newUser.user.uid}`), {
        last_login: dt,
      }); */
      onNavigate('/home');

      showMessage(`Welcome ${newUser.user.email}`, 'success');
    } catch (error) {
      if (error.code === 'auth/wrong-password') {
        showMessage('Wrong password', 'error'); // despues de la coma viene el tipo (estilo que le cambia el color al msg)
      } else if (error.code === 'auth/user-not-found') {
        showMessage('User not found', 'error');
      } else if (error.code) {
        showMessage(error.message, 'error');
      }
    }
  });

  /* Google log-in */
  const googleBtn = section.querySelector('#googleLogin');
  googleBtn.addEventListener('click', async (e) => {
    e.preventDefault();
    try {
      await loginGoogle();
      onNavigate('/profile');
      showMessage(`Welcome ${loginGoogle.user.displayName}`, 'success');
    } catch (error) {
      if (error.code) {
        showMessage(error.message, 'error');
      }
    }
  });

  /* Facebook log-in */
  const facebookbtn = section.querySelector('#fb-login');
  facebookbtn.addEventListener('click', async (e) => {
    e.preventDefault();

    try {
      await loginFacebook();
      onNavigate('/profile');
      showMessage(`Welcome ${loginFacebook.user.displayName}`, 'success');
    } catch (error) {
      if (error.code) {
        showMessage(error.message, 'error');
      }
    }
  });

  /* AGREGAR TEXTO A LOS BOTONES textContent */
  button.textContent = 'Entrar';
  buttonBack.textContent = 'Regresa';
  buttonProfile.textContent = 'Profile';
  title.textContent = 'Inicia secion LOGIN';

  button.addEventListener('click', () => {
    onNavigate('/home');
  });
  buttonBack.addEventListener('click', () => {
    onNavigate('/');
  });
  buttonProfile.addEventListener('click', () => {
    onNavigate('/profile');
  });

  /* INSERTA append */
  div.append(title, buttonProfile, button, buttonBack, section);

  return div;
};
