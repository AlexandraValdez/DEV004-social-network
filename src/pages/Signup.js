// import { getDatabase, set, ref } from 'firebase/database';
import { onNavigate } from '../router.js';
import { addUser, loginGoogle, loginFacebook } from '../firebase/auth.js';
import { showMessage } from '../components/showMessage.js';

export const signUp = () => {
  const div = document.createElement('div');
  const title = document.createElement('h2');
  const buttonBack = document.createElement('button');
  const section = document.createElement('div');
  section.innerHTML = `
    <article class="wrapper">
      <picture class="img-container">
        <img src="./images/Sonder-icon.png" alt="">
      </picture>
      <h1 class="h1-form">Sign Up</h1>
      <p class="text-sign">Find and share inspiration all around the world!</p>
      <form id="signup-form" >
        <input id="singup-name" type="text" class="" placeholder="Name">
        <input id="singup-lastname" type="text" class="" placeholder="Last Name">
        <input id="singup-email" type="email" class="" placeholder="Email">
        <input id="singup-password" type="password" class="" placeholder="Password">
        <input type="password" class="" placeholder="Confirm Password">
      <button type="submit" class="submit">Sign Up</button>
      </form>  
      <p class="or">or</p>
        <button id="fb-login" type="button" class="submit facebook">
        <img src="./images/facebook.png" alt="facebook icon">Continue with Facebook</button>
        <button id="googleLogin" type="button" class="submit google">
        <img src="./images/google.png" alt="google icon">Continue with Google</button>
        <p class="p-log">
          Already have an account yet? <a class="link" href="/login">Log in</a>
        </p> 
    </article>
    `;
  /* FIREBASE FORM */
  const signupForm = section.querySelector('#signup-form');
  signupForm.addEventListener('submit', async (e) => {
    e.preventDefault();

    const username = signupForm['singup-name'].value;
    const lastname = signupForm['singup-lastname'].value;
    const email = signupForm['singup-email'].value;
    const password = signupForm['singup-password'].value;
    // eslint-disable-next-line no-console
    console.log(username, lastname, email, password);

    try {
      /* set(ref(database, `users/${addUser.user.uid}`), {
        name: username,
        lastname,
        email,
      }); */
      addUser(email, password);
      onNavigate('/profile');
      showMessage(`Welcome ${addUser.user.email}`, 'success');
    } catch (error) {
      if (error.code === 'auth/email-already-in-use') {
        showMessage('Email already in use', 'error'); // despues de la coma viene el tipo (estilo que le cambia el color al msg)
      } else if (error.code === 'auth/invalid-email') {
        showMessage('Invalid email', 'error');
      } else if (error.code === 'auth/weak-password') {
        showMessage('Weak password', 'error');
      } else if (error.code) {
        showMessage(error.message, 'error');
      }
    }
  });

  /* Google log-in */
  const googleBtn = section.querySelector('#googleLogin');
  googleBtn.addEventListener('click', async (e) => {
    e.preventDefault();

    try {
      // setDoc(ref(database, `users/${addUser.user.uid}`), {
      //   name: username,
      //   lastname,
      //   email,
      // });
      await loginGoogle();
      onNavigate('/home');
      showMessage(`Welcome ${loginGoogle.user.displayName}`, 'success');
    } catch (error) {
      if (error.code === 'auth/email-already-in-use') {
        showMessage('Email already in use', 'error'); // despues de la coma viene el tipo (estilo que le cambia el color al msg)
      } else if (error.code === 'auth/invalid-email') {
        showMessage('Invalid email', 'error');
      } else if (error.code === 'auth/weak-password') {
        showMessage('Weak password', 'error');
      } else if (error.code) {
        showMessage(error.message, 'error');
      }
    }
  });

  /* Facebook log-in */
  const facebookbtn = section.querySelector('#fb-login');
  facebookbtn.addEventListener('click', async (e) => {
    e.preventDefault();

    try {
      await loginFacebook();
      onNavigate('/home');
      showMessage(`Welcome ${loginFacebook.user.displayName}`, 'success');
    } catch (error) {
      if (error.code) {
        showMessage(error.message, 'error');
      }
    }
  });
  // button.textContent = "crear cuenta";
  buttonBack.textContent = 'Regresa';
  // title.textContent = "Registro de nueva cuenta";

  /* evento a boton */
  // button.addEventListener("click", () => {
  //   onNavigate("/");
  // });
  buttonBack.addEventListener('click', () => {
    onNavigate('/');
  });

  /* INSERTA append */
  div.append(title, buttonBack, section);

  return div;
};
// inputEmail, inputPass, button,
