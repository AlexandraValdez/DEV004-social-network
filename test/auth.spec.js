/**
 * @jest-environment jsdom
 */

// importamos la funcion que vamos a testear
import { signUp } from '../src/pages/Signup.js';
import { Login } from '../src/pages/Login.js';
import { login } from '../src/firebase/auth.js';

jest.mock('../src/router.js');
jest.mock('../src/firebase/auth.js');

describe('myFunction', () => {
  it('debería ser una función', () => {
    expect(typeof signUp).toBe('function');
  });
});

describe('Login', () => {
  it('deberia mostrar el formulario de login correctamente', () => {
    const root = document.createElement('div');
    root.id = 'root';

    root.append(Login());

    const loginForm = root.querySelector('#login-form');
    const email = root.querySelector('#login-email');
    const password = root.querySelector('#login-password');
    const submit = root.querySelector('#log');

    expect(loginForm).not.toBe(null);
    expect(email).not.toBe(null);
    expect(password).not.toBe(null);
    expect(submit).not.toBe(null);

    document.body.append(root);
  });
});

describe('Los test del Registro', () => {
  test('Existe el boton de crear cuenta', () => {
    const elemento = Login();
    const boton = elemento.querySelector('#log');
    expect(boton).not.toBeNull();
  });

  test('Snapshot del REGISTRO', () => {
    expect(login()).toMatchSnapshot();
  });
});
