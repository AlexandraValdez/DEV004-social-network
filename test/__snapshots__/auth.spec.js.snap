// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Los test del Registro Snapshot del REGISTRO 1`] = `undefined`;
